smartsim/experiment.py

```diff
@@ -737,6 +751,9 @@ def create_database(
         :return: Orchestrator
         :rtype: Orchestrator or derived class
         """
+
+        self.append_to_db_identifier_list(db_identifier)
```

Comment:
the id list feels like it duplicates information found elsewhere (e.g. the list of orchestrators) and gives us two sources of truth that can fall out of sync. Consider not adding this additional collection.